--- a/cmake/oskar_compiler_options.cmake
+++ b/cmake/oskar_compiler_options.cmake
@@ -248,6 +248,9 @@
         elseif (ARCH MATCHES 8.7)
             message("-- INFO: Building CUDA device code for architecture 8.7")
             list(APPEND CUDA_NVCC_FLAGS -gencode arch=compute_87,code=sm_87)
+        elseif (ARCH MATCHES 9.0)
+            message("-- INFO: Building CUDA device code for architecture 9.0")
+            list(APPEND CUDA_NVCC_FLAGS -gencode arch=compute_90,code=sm_90)
         else()
             message(FATAL_ERROR "-- CUDA_ARCH ${ARCH} not recognised!")
         endif()

